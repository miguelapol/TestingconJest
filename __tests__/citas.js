import Citas from '../js/classes/Citas';
//aqui comprobamos la funcionalidad de la clase
describe('Probar la clase de citas',()=>{
    const citas= new Citas();
    const id= Date.now();
    test('Agregar nuevas funciones',()=>{
        const citaObj={
            id,
            mascota:'Hook',
            propietario: 'Miguel',
            telefono: '55986628',
            fecha: '10-12-2021',
            hora: '10:30',
            sintomas: 'Solo duerme'
        };

        citas.agregarCita(citaObj);
        //pruebas verificamos que la clase que pasamos se esta Â´pasando correctamente
        expect(citas).toMatchSnapshot();
    });
    test('Actualizar cita',()=>{
        const citaActualizada={
            id,
            mascota:'Negrito',
            propietario: 'Miguel',
            telefono: '55986628',
            fecha: '10-12-2021',
            hora: '10:30',
            sintomas: 'Solo duerme'
        };
        citas.editarCita(citaActualizada);
        //prueba
        expect(citas).toMatchSnapshot();
    })
    test('Eliminar cita',()=>{
        citas.eliminarCita(id);

        //prueba aqui debe estar el arreglo vacio
        expect(citas).toMatchSnapshot();
    });
 
})