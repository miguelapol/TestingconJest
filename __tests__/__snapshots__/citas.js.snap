// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase de citas Actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2021",
      "hora": "10:30",
      "id": 1655918218170,
      "mascota": "Negrito",
      "propietario": "Miguel",
      "sintomas": "Solo duerme",
      "telefono": "55986628",
    },
  ],
}
`;

exports[`Probar la clase de citas Agregar nuevas funciones 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2021",
      "hora": "10:30",
      "id": 1655918218170,
      "mascota": "Hook",
      "propietario": "Miguel",
      "sintomas": "Solo duerme",
      "telefono": "55986628",
    },
  ],
}
`;

exports[`Probar la clase de citas Eliminar cita 1`] = `
Citas {
  "citas": Array [],
}
`;
